hyper/http20/response.py

```diff
@@ -71,12 +77,13 @@ def __init__(self, headers, stream):
         # This 16 + MAX_WBITS nonsense is to force gzip. See this
         # Stack Overflow answer for more:
         # http://stackoverflow.com/a/2695466/1401686
-        if b'gzip' in self.headers.get(b'content-encoding', []):
-            self._decompressobj = zlib.decompressobj(16 + zlib.MAX_WBITS)
-        elif b'deflate' in self.headers.get(b'content-encoding', []):
-            self._decompressobj = DeflateDecoder()
-        else:
-            self._decompressobj = None
+        self._decompressobj = next(
+            (
+                decompressors.get(c)
+                for c in self.headers.get(b'content-encoding', [])
+            ),
+            None
```

Comment:
So at this point I think this is still less clear than just writing a `for` loop with a `break` in it.